regression/cbmc-primitives/pointer-offset-01/test-no-cp.desc

```diff
@@ -1,4 +1,4 @@
-CORE
+CORE new-smt-backend
```

Comment:
Please fold updates to test into the commit that fixes the problem. Otherwise it is very hard to understand what effect a change has.

Reply:
I can, however the third commit contains regression tests that need the second commit fix to work.
So I decided to keep 
 - Commit 1: add regression tests that were working because of previous PRs (see other comment)
 - Commit 2: fix issue with unsigned pointer offset
 - Commit 3: add regression tests now working because of commit 2